In [2]:
import pandas as pd
import string
import json

In [28]:
anew_w = pd.read_csv("anew.csv")
anew_w = anew_w.drop(columns=["arousal", "dominance"])

# print([anew_w[anew_w["term"] == "violent"]])

def post_analyzer(text = str):
    new_text = text.translate(str.maketrans("", "", string.punctuation))
    new_text = new_text.lower().split()

    positive_w = []
    p_sum = 0
    positive_score = 0

    negative_w = []
    n_sum = 0
    negative_score = 0

    for word in new_text:
        if word in anew_w["term"].values:
            element = anew_w[anew_w["term"] == word]
            pleasure = element["pleasure"].iloc[0]
            score = (pleasure - 50)/50 * 10

            if score > 0:
                p_sum += score
                positive_w.append((word, score))

            elif score < 0:
                n_sum += score
                negative_w.append((word, score))

    # First method: Simple comparison between the positive and negative score
    result = p_sum - (-n_sum)

    # Second method: When there is a big difference between the number of positive and negative words
    # if len(positive_w) > 0:
    #     positive_score = p_sum/(len(positive_w) * 10)

    # if len(negative_w) > 0:
    #     negative_score = -n_sum/(len(negative_w) * 10)

    # result = positive_score - negative_score

    if result > 0:
        return "Positive post"
    return "Negative post"

In [29]:
text = '''
I walked into Spider-Man: Brand New Day expecting another fun MCU movie, but I left feeling like I'd just watched one of the best Spider-Man films ever made. It finally feels like Peter Parker has earned the title of Spider-Man. He's alone, nobody remembers who he is after No Way Home, and you can feel the weight of that in every decision he makes. What makes it so good is that it goes back to the character's roots. Peter isn't relying on Avengers or high-tech gadgets-he's just a trying to protect New York while dealing with grief, loneliness, and impossible choices. It feels grounded in a way the previous films sometimes didn't and has a darkness to it and you sympathise with it. Spider-Man movies never fail to miss.
'''

print(post_analyzer(text))

Positive post


In [30]:
analyze_result = {}

with open("imdb_tt22084616_reviews.json", "r", encoding="utf-8") as post_file:
    posts = json.load(post_file)

for post in posts["reviews"]:
    analyze_result[post["review_id"]] = post_analyzer(post["review"])

print(analyze_result)


{'rw11707575': 'Positive post', 'rw11707705': 'Positive post', 'rw11707560': 'Positive post', 'rw11708461': 'Positive post', 'rw11709320': 'Positive post', 'rw11708113': 'Positive post', 'rw11707902': 'Positive post', 'rw11711838': 'Positive post', 'rw11707859': 'Positive post', 'rw11718837': 'Positive post', 'rw11712219': 'Positive post', 'rw11709354': 'Positive post', 'rw11744024': 'Positive post', 'rw11725956': 'Positive post', 'rw11709536': 'Positive post', 'rw11713717': 'Positive post', 'rw11714819': 'Positive post', 'rw11751085': 'Positive post', 'rw11713664': 'Positive post', 'rw11727577': 'Positive post', 'rw11708560': 'Positive post', 'rw11708082': 'Positive post', 'rw11715517': 'Positive post', 'rw11709397': 'Positive post', 'rw11727676': 'Positive post', 'rw11707615': 'Positive post', 'rw11708790': 'Positive post', 'rw11760394': 'Positive post', 'rw11714426': 'Positive post', 'rw11727614': 'Positive post', 'rw11710435': 'Positive post', 'rw11745108': 'Negative post', 'rw1172

In [31]:
positive = 0
negative = 0

for result in analyze_result:
    if analyze_result[result] == "Positive post":
        positive += 1

    else:
        negative += 1

# with open("results.txt", "w") as results_file:
#     results_file.write(f"")

#     results_file.write(f"Positive posts: {positive/(positive + negative) * 100}%")
#     results_file.write(f"Negative posts: {negative/(positive + negative) * 100}%")

print(f"Positive posts: {positive/(positive + negative) * 100}%")
print(f"Negative posts: {negative/(positive + negative) * 100}%")

Positive posts: 97.6%
Negative posts: 2.4%


## Results obtained form the first method
- Positive posts: 97.6%
- Negative posts: 2.4%

## Results obtained from the second method
- Positive posts: 93.4%
- Negative posts: 6.6000000000000005%